### PR TITLE
Make tcencode more performant by avoiding json roundtrip conversion

### PR DESCRIPTION
**Pull request by patmmccann**
fixes https://github.com/InteractiveAdvertisingBureau/iabtcf-es/issues/444

### CODE REVIEW

**Comment by morinel** [comment]
Thank you, a N=1 preliminary test shows this already greatly improves performance, reducing the time to encode the string from 75ms to 15ms. If the cloning of the GVL could be done only once, this would once again halve the time spent, to maybe 8ms.

File: modules/core/src/GVL.ts [hunk 1]
```diff
@@ -401,21 +401,33 @@ export class GVL extends Cloneable<GVL> implements VendorList {
...
+        tcfPolicyVersion,
+        lastUpdated,
+        purposes,
+        specialPurposes,
+        features,
+        specialFeatures,
+        stacks,
+        dataCategories,
+        vendors: fullVendorList
+    };
```
Comment (sevriugin) [inline modules/core/src/GVL.ts:430]: This is not a deep clone of `VendorList`, right? We will return references to objects like `vendors` and anyone outside of the library could update  it, maybe it's not good. 

The `JSON.parse(JSON.stringify())` is ugly and very slow particular for the very big object like GVL. Maybe we can develop an alternative solution creating custom deep clone method in this case as we know the GVL structure?

**Comment by sevriugin** [comment]
There is [PR](https://github.com/InteractiveAdvertisingBureau/iabtcf-es/pull/447) for GVL clone avoiding  `JSON.parse(JSON.stringify())`

**Comment by HeinzBaumann** [comment]
I suggest we close this PR in favor of https://github.com/InteractiveAdvertisingBureau/iabtcf-es/pull/447

**Comment by sevriugin** [comment]
I have tested the new clone vs old one and it works 40% faster:
`new: 0,340041995048523`
`old: 0,545000076293945`
we can think about more performance improvements but as the the first step it's good enough

**Pull request closed without merging**